Parsers/ASimAuthentication/Parsers/vimAuthenticationGithub.yaml

```diff
@@ -0,0 +1,133 @@
+Parser:
+  Title: ASIM Authentication filtering parser for Github
+  Version: "0.1.0"
+  LastUpdated: Apr 12, 2024
+Product:
+  Name: Github
+Normalization:
+  Schema: Authentication
+  Version: "0.1.3"
+References:
+  - Title: Using functions
+    Link: https://docs.microsoft.com/azure/azure-monitor/log-query/function
+  - Title: ASIM Authentication Schema
+    Link: https://aka.ms/ASimAuthenticationDoc
+  - Title: ASIM
+    Link: https:/aka.ms/AboutASIM
+  - Title: Github
+    Link: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/security-log-events
+Description: |
+  This ASIM parser supports filtering and normalizing the Github logs stored in 'GitHubAuditLogPolling_CL' table to the ASIM authentication normalized schema.
+ParserName: vimAuthenticationGithub
+EquivalentBuiltInParser: _Im_Authentication_Github
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: actor_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: targetappname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: srchostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventtype_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+     let parser = (
+     starttime:datetime=datetime(null), 
+     endtime:datetime=datetime(null), 
+     username_has_any:dynamic = dynamic([]),
+     targetappname_has_any:dynamic = dynamic([]),
+     srcipaddr_has_any_prefix:dynamic = dynamic([]),
+     srchostname_has_any:dynamic = dynamic([]),
+     eventtype_in:dynamic =dynamic([]),
+     eventresultdetails_in:dynamic = dynamic([]),
+     eventresult:string = '*',
+     disabled:bool=false
+     )
+     {
+     let EventResultDetailslookup=datatable(EventOriginalResultDetails:string,EventResult:string,EventType:string,EventSeverity:string )
+     ["user.login","Success","logon","Informational",
+     "user.logout","Success","logoff","Informational",
+     "user.two_factor_challenge_success","Success","logon","Informational",
+     "user.two_factor_challenge_failure","Failure","logon","Informational",
+     "user.two_factor_requested","Success","logon","Informational",
+     "user.failed_login","Failure","logon","Informational",
+     "user.sign_in_from_unrecognized_device_and_location","Success","logon","High",
+     "user.sign_in_from_unrecognized_device","NA","logon","High",
```

Comment:
@siteshjha11 
we are setting sign_in_from_unrecognized_device EventResult as NA and sign_in_from_unrecognized_device_and_location as Success. Any specific reason one is NA and other is Success